android-components/components/support/remotesettings/src/test/java/mozilla/components/support/remotesettings/RemoteSettingsClientTest.kt

```diff
@@ -114,4 +115,38 @@ class RemoteSettingsClientTest {
             assertEquals(fetchResult, result)
         }
     }
+
+    @Test
+    fun `GIVEN existing file with valid content WHEN read is called THEN return Success`() {
+        val tempFile = File.createTempFile("test", ".json")
+        val validJsonString = """
+            {
+                "response": {
+                    "records": [
+                        {"id": "1", "lastModified": 123, "deleted": false, "attachment": null, "fields": {}}
+                    ],
+                    "lastModified": 123
+                }
+            }
+        """.trimIndent()
+        val remoteSettingsClient = RemoteSettingsClient(
+            tempFile,
+            "https://example.com",
+            "apiKey",
+            "collectionName",
+        )
+
+        tempFile.writeText(validJsonString)
+        remoteSettingsClient.file = tempFile
+
+        val result = runBlocking { remoteSettingsClient.read() }
+
+        assertEquals(true, tempFile.exists())
+        assertEquals(validJsonString, tempFile.readText())
+
+        assertTrue(result is RemoteSettingsResult.Success)
+        assertEquals(123uL, (result as RemoteSettingsResult.Success).response.lastModified)
```

Comment:
Ah, I see that changing this test to this:
```
assertEquals(1, (result as RemoteSettingsResult.Success).response.records.first().id)
```
results in a casting exception. That explains the changes above, but it does mean that we will be causing crashes at runtime if we land this patch. I think we'll need to add a more robust (de)serialization method in order to get this working as intended.